mantid-developer-osx.yml

```diff
@@ -42,8 +42,7 @@ dependencies:
   - setuptools=47.0.0 # Pinned purposefully due to incompatibility with later versions
   - sphinx >= 4.5.*
   - sphinx_bootstrap_theme=>0.8.1
-  - tbb-devel=2020.2.*
```

Comment:
I've take out `tbb` here as `tbb-devel` will depend on it so we reduce the duplication by just allowing `tbb` to follow `tbb-devel`